koschei/backend/__init__.py

```diff
@@ -112,12 +116,17 @@ def submit_build(session, package):
     )
     if srpm_res:
         srpm, srpm_url = srpm_res
+        if session.build_from_repo_id:
+            target = None
+            build_opts.update({'repo_id': package.collection.latest_repo_id})
```

Comment:
By benefits I mean it can be resolved immediately, it doesn't need to wait for polling to fetch the repo_id from koji